sbom/bom-go-mod.spdx

```diff
@@ -0,0 +1,1299 @@
+SPDXVersion: SPDX-2.2
```

Comment:
Is this something we want to store in Git? This seems more like a build artifact that can be stored in Jenkins or Nexus. I would suggest adding the `sbom` directory to `.gitignore`

Reply:
good question. same as @bnevis-i brought it up in https://github.com/edgexfoundry/edgex-go/pull/4339#issuecomment-1424604298 .    @bnevis-i  Probably we can discuss and decide this in the security WG meeting.

Reply:
this file is removed and add it into .gitignore